opsdroid/connector/twitch/__init__.py

```diff
@@ -0,0 +1,769 @@
+
+        Args:
+            topic (string): Twitch webhook url to subscribe/unsubscribe to.
+            mode (string): subscribe or unsuscribe to the webhook.
+
+        """
+        _LOGGER.info(_("Attempting to connect to webhook %s."), topic)
+
+        if topic == "follows":
+            topic = f"{TWITCH_API_ENDPOINT}/users/follows?to_id={self.user_id}&first=1"
+
+        if topic == "stream changed":
+            topic = f"{TWITCH_API_ENDPOINT}/streams?user_id={self.user_id}"
+
+        if topic == "subscribers":
+            topic = f"{TWITCH_API_ENDPOINT}/subscriptions/events?broadcaster_id={self.user_id}&first=1"
+
+        headers = {"Client-ID": self.client_id, "Authorization": f"Bearer {self.token}"}
+
+        async with aiohttp.ClientSession() as session:
+
+            payload = {
+                "hub.callback": f"{self.base_url}/connector/{self.name}",
+                "hub.mode": mode,
+                "hub.topic": topic,
+                "hub.lease_seconds": 60 * 60 * 24 * 9,  # Expire after 9 days
+                "hub.secret": self.webhook_secret,
+            }
+
+            response = await session.post(
+                TWITCH_WEBHOOK_ENDPOINT, headers=headers, json=payload
+            )
+
+            if response.status >= 400:
+                _LOGGER.debug(_("Error: %s - %s"), response.status, response.text)
+
+    async def handle_challenge(self, request):
+        """Challenge handler for get request made by Twitch.
+
+        Upon subscription to a Twitch webhook, Twitch will do a get request to the
+        `callback` url provided to check if the url exists. Twitch will do a get request
+        with a challenge and expects the `callback` url to return that challenge in plain-text
+        back to Twitch.
+
+        This is what we are doing here, we are getting `hub.challenge` from the request and return
+        it in plain-text, if we can't find that challenge we will return a status code 500.
+
+        Args:
+            request (aiohttp.web.Request): Request made to the get route created for webhook subscription.
+
+        Returns:
+           aiohttp.web.Response: if request contains `hub.challenge` we return it, otherwise return status 500.
+
+        """
+        challenge = request.rel_url.query.get("hub.challenge")
+
+        if challenge:
+            return aiohttp.web.Response(text=challenge)
+
+        _LOGGER.debug(_("Failed to get challenge from GET Request made by Twitch."))
+        return aiohttp.web.Response(status=500)
+
+    async def twitch_webhook_handler(self, request):
+        """Handle event from Twitch webhooks.
+
+        This method will handle events when they are pushed to the webhook post route. Each webhook will
+        send a different kind of payload so we can handle each event and trigger the right opsdroid event
+        for the received payload.
+
+        For follow events the payload will contain `from_id`(broadcaster id), `from_username`(broadcaster username)
```

Comment:
@jacobtomlinson is correct these need to be double backticks, also I would put a space after the close backticks.